server/src/main/java/org/elasticsearch/cluster/routing/UnassignedInfo.java

```diff
@@ -366,8 +394,17 @@ public AllocationStatus getLastAllocationStatus() {
      *
      * @return calculated delay in nanoseconds
      */
-    public long getRemainingDelay(final long nanoTimeNow, final Settings indexSettings) {
-        long delayTimeoutNanos = INDEX_DELAYED_NODE_LEFT_TIMEOUT_SETTING.get(indexSettings).nanos();
+    public long getRemainingDelay(
+        final long nanoTimeNow,
+        final Settings indexSettings,
+        final Map<String, SingleNodeShutdownMetadata> nodesShutdownMap
+    ) {
+        long delayTimeoutNanos = Optional.ofNullable(lastAllocatedNodeId)
+            .map(nodesShutdownMap::get)
+            .filter(shutdownMetadata -> SingleNodeShutdownMetadata.Type.RESTART.equals(shutdownMetadata.getType()))
```

Comment:
It is a bit odd that if a node crashes and then has a shutdown indication added, we extend the node left timeout to be the possibly longer shutdown node left timeout.

I wonder if we could instead capture whether the node was ready for shutdown when we lost it and use that to determine which of the delays to use?

Reply:
See my above comments for the difficulty with doing so, but as an alternative, we could at least capture whether the node was registered for shutdown at all when the node went offline.

Reply:
Yes, given your earlier comments, that approach makes sense to me.